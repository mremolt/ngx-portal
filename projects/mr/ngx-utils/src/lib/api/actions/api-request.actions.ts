import { HttpErrorResponse, HttpRequest } from '@angular/common/http';
import { Action } from '@ngrx/store';

import { ApiActions, ApiRequestConfig } from '../types';

export enum ApiRequestActionTypes {
  ApiRequest = '[API] Request',
  ApiError = '[API] Error',
}

export class ApiRequest<T = any> implements Action {
  public readonly type = ApiRequestActionTypes.ApiRequest;

  constructor(public payload: ApiRequestConfig<T>) {}
}

export class ApiError implements Action {
  public readonly type = ApiRequestActionTypes.ApiError;

  constructor(public payload: HttpErrorResponse) {}
}

export class ApiGetAction<T> extends ApiRequest {
  constructor(url: string, actions: ApiActions<T>) {
    const request = new HttpRequest<T>('GET', url);
    super({ request, actions });
  }
}

export class ApiPostAction<T> extends ApiRequest {
  constructor(url: string, body: T | null, actions: ApiActions<T>) {
    const request = new HttpRequest<T>('POST', url, body);
    super({ request, actions });
  }
}

export class ApiPutAction<T> extends ApiRequest {
  constructor(url: string, body: T | null, actions: ApiActions<T>) {
    const request = new HttpRequest<T>('PUT', url, body);
    super({ request, actions });
  }
}

export class ApiDeleteAction<T> extends ApiRequest {
  constructor(url: string, actions: ApiActions<T>) {
    const request = new HttpRequest<T>('DELETE', url);
    super({ request, actions });
  }
}
