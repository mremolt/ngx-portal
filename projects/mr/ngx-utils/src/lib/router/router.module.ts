import { NgModule } from '@angular/core';
import { EffectsModule } from '@ngrx/effects';
import { RouterEffects } from './router.effects';

@NgModule({
  imports: [EffectsModule.forFeature([RouterEffects])],
})
export class RouterModule {}
