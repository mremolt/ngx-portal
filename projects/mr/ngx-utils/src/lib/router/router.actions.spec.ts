import { Back, Forward, Go } from './router.actions';

describe('Router actions', () => {
  describe('Go', () => {
    it('creates the correct action', () => {
      expect(new Go({ path: ['foo'] })).toMatchSnapshot();
    });
  });

  describe('Back', () => {
    it('creates the correct action', () => {
      expect(new Back()).toMatchSnapshot();
    });
  });

  describe('Forward', () => {
    it('creates the correct action', () => {
      expect(new Forward()).toMatchSnapshot();
    });
  });
});
