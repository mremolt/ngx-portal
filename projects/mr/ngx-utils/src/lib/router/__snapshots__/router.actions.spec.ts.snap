// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Router actions Back creates the correct action 1`] = `
Back {
  "type": "[Router] Back",
}
`;

exports[`Router actions Forward creates the correct action 1`] = `
Forward {
  "type": "[Router] Forward",
}
`;

exports[`Router actions Go creates the correct action 1`] = `
Go {
  "payload": Object {
    "path": Array [
      "foo",
    ],
  },
  "type": "[Router] Go",
}
`;
